solutions/instances/variables.tf

```diff
@@ -0,0 +1,287 @@
+########################################################################################################################
+# Common variables
+########################################################################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The API Key to use for IBM Cloud."
+  sensitive   = true
+}
+
+variable "existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group in which to provision resources to."
+}
+
+variable "region" {
+  description = "Region where observability resources will be created"
+  type        = string
+  default     = "us-south"
+}
+
+##############################################################################
+# Log Analysis Variables
+##############################################################################
+
+variable "log_analysis_instance_name" {
+  type        = string
+  description = "The name of the IBM Cloud Logging instance to create."
+  default     = "log-analysis"
+}
+
+variable "log_analysis_plan" {
+  type        = string
+  description = "The IBM Cloud Logging plan to provision. Available: lite, 7-day, 14-day, 30-day, hipaa-30-day"
+  default     = "7-day"
+
+  validation {
+    condition     = can(regex("^lite$|^7-day$|^14-day$|^30-day$|^hipaa-30-day$", var.log_analysis_plan))
+    error_message = "The log_analysis_plan value must be one of the following: lite, 7-day, 14-day, 30-day, hipaa-30-day."
+  }
+}
+
+variable "log_analysis_service_endpoints" {
+  description = "The type of the service endpoint that will be set for the Log Analysis instance."
+  type        = string
+  default     = "private"
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.log_analysis_service_endpoints)
+    error_message = "The specified service_endpoints is not a valid selection"
+  }
+}
+
+variable "log_analysis_tags" {
+  type        = list(string)
+  description = "Tags associated with the IBM Cloud Logging instance (Optional, array of strings)."
+  default     = []
+}
+
+variable "enable_archive" {
+  type        = bool
+  description = "Enable archive on log analysis instance"
+  default     = true
+}
+
+variable "archive_api_key" {
+  type        = string
+  description = "(Optional) The IBM Cloud API key will be used for Log Analysis archiving to COS if this is not passed."
```

Comment:
Its not clear here that the api key value passed in the `ibmcloud_api_key` variable will be used if no value is passed here